gammapy/background/models.py

```diff
@@ -24,6 +24,91 @@
 DEFAULT_SPLINE_KWARGS = dict(k=1, s=0)
 
 
+def _select_closest():
+    pass
+
+
+def add_column_and_sort_table(sources, pointing_position):
```

Comment:
The `sources.sort` call means that the table a user puts in will be modified.

It's usually better to not modify inputs, but only to compute something and return a new object.
If functions modify inputs, that's often surprising and hard to debug for users / callers.

So in this case, you should add a line that makes a copy of the input table at the top:
```python
sources = sources.copy()
```